### ISSUE TITLE
regarding the bleu score reported

### ISSUE THREAD

**Issue by fawazsammani** [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning#17]
Hi. In the paper, they have reported the blue-4 score as 25. Is this reported score in the paper obtained without using teacher forcing during evaluation? 
You mentioned that you have achieved a score of 24.29 with teacher forcing during evaluation, and then you reported 32.95 without teacher forcing (and with using beam search of 3), and that is higher than the paper. 
So is the score reported in the paper with or without using teacher forcing?

**Comment by sgrvinod** [comment]
I think the paper reports it without teacher forcing, as it is meant to be. The original Theano repo used the [cross entropy loss as the validation metric](https://github.com/kelvinxu/arctic-captions/issues/5#issuecomment-109019054), so the BLEU must have been reported in a natural inference setting. To me, it doesn't make a lot of sense to report the BLEU on teacher-forced captions as it doesn't really reflect performance.

About the difference in the scores, I'm not really sure. I've speculated -

> This is higher than the result in the paper, and could be because of how our BLEU calculators are parameterized, the fact that I used a ResNet encoder, and actually fine-tuned the encoder – even if just a little.

**Comment by mitjanikolaus** [comment]
As a follow-up question regarding the BLEU scores reported in the README here:

How can it be that the BLEU score calculated using eval.py, without teacher forcing, beam size 1 is significantly higher (29.98) than the score achieved during training of the model (apparently only 24.29 after fine-tuning)? Shouldn't it be comparable or even lower, because no teacher forcing happens during evaluation?

**Comment by kmario23** [comment]
@mitjanikolaus beam size 1 means we simply take the predicted word with highest score as the next word. This is sometimes called greedy decoding.
The score obtained with not using teacher forcing will be mostly higher because that's where the magic of beam search comes into play. Ideally, one should use a beam size of 2 or higher for better accuracy.
